src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -239,6 +249,21 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
             factoryContext.TrackedParameters.Add(parameter.Name, RequestDelegateFactoryConstants.BodyAttribute);
             return BindParameterFromBody(parameter, bodyAttribute.AllowEmpty, factoryContext);
         }
+        else if (parameterCustomAttributes.OfType<IFromFormMetadata>().FirstOrDefault() is { } formAttribute)
+        {
+            if (parameter.ParameterType == typeof(IFormFileCollection))
+            {
+                if (!string.IsNullOrEmpty(formAttribute.Name))
+                {
+                    throw new NotSupportedException(
+                        $"Assigning a value to the {nameof(IFromFormMetadata)}.{nameof(IFromFormMetadata.Name)} property is not supported for parameters of type {nameof(IFormFileCollection)}.");
```

Comment:
Do we throw any startup errors when this attribute is applied to parameters that aren't either an `IFormFile` or `IFormFileCollection`?

Reply:
I'll check - assuming not, shall I add a catch-all for any other type than the two new ones?

Reply:
Sounds good!

Reply:
The current behaviour is the below, but I've added a special case that throws `NotSupportedException` instead.

```
typeof(System.ArgumentException): Expression of type 'Microsoft.AspNetCore.Http.IFormFile' cannot be used for assignment to type 'System.Boolean'
```